Core/Frameworks/Flake/Framework.php

```diff
@@ -273,8 +273,10 @@ protected static function initDb() {
         if (defined("BAIKAL_CONTEXT_INSTALL") && (!isset($config['system']['configured_version']) || $config['system']['configured_version'] === BAIKAL_VERSION)) {
             return true;
         }
-        if ($config['database']['mysql'] === true) {
+        if ($config['database']['backend'] === 'mysql') {
             self::initDbMysql($config);
+        } elseif ($config['database']['backend'] === 'pgsql') {
+            self::initDbPgsql($config);
```

Comment:
Apparently, Flake (the framework Baikal is built on) connects to the database and runs this code before going through the migration steps. This makes it a bit hard to do the migration properly. The problem is that we already need the database connection in the earlier steps of `VersionUpgrade` in case someone upgrades multiple releases at once. I think the easiest option (though not 100% elegant) would be to check both the old and the new config format here and choose `backend` only if it exists.

Reply:
I see, that seems reasonable, implemented in 35a3d01 ([Compare](https://github.com/sabre-io/Baikal/compare/ef019bcfd407ca32f6845d9384e1919bd4d772bc..35a3d018f135ab2a686206be20475cc792cf4596#diff-ee2f5d9758e05e670b748560a31065fd3a0fbf1ead17de608e45bc9a4594a214R277))